src/zarr/store/zip.py

```diff
@@ -0,0 +1,244 @@
+from __future__ import annotations
+
+import os
+import threading
+import time
+import zipfile
+from collections.abc import AsyncGenerator
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Literal
+
+from zarr.abc.store import Store
+from zarr.core.buffer import Buffer, BufferPrototype
+
+if TYPE_CHECKING:
+    from typing_extensions import Self
+
+ZipStoreAccessModeLiteral = Literal["r", "w", "a"]
+
+
+class ZipStore(Store):
```

Comment:
Just a note that we should have nice docstrings for the store classes, and their methods. This is not a blocker, and can be handled by a later PR.